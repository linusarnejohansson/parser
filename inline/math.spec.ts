import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("inline.math", () => {
	it("basic", () => {
		const result = parse(new Source(IO.StringReader.create("$a^2 + b^2 = c^2$"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("in text", () => {
		const result = parse(new Source(IO.StringReader.create("This is a text with an $a^2 + b^2 = c^2$ in it."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
