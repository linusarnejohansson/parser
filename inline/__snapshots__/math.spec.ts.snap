// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline.math basic 1`] = `
Array [
  Object {
    "class": "Inline.Math",
    "value": "a^2 + b^2 = c^2",
  },
]
`;

exports[`inline.math in text 1`] = `
Array [
  Object {
    "class": "Inline.Text",
    "value": "This is a text with an ",
  },
  Object {
    "class": "Inline.Math",
    "value": "a^2 + b^2 = c^2",
  },
  Object {
    "class": "Inline.Text",
    "value": " in it.",
  },
]
`;
